include/nbl/system/IFileArchive.h

```diff
@@ -72,16 +76,17 @@ class IFileArchive : public core::IReferenceCounted
 	//! Opens a file based on its name
 	virtual core::smart_refctd_ptr<IFile> readFile(const SOpenFileParams& params) = 0;
 
-	const core::vector<SFileListEntry>& getFiles() const { return m_files; }
+	const core::vector<SFileListEntry>& getArchivedFiles() const { return m_files; }
+	IFile* asFile() { return m_file.get(); }
 protected:
-	virtual void addItem(const system::path& fullPath, uint32_t offset, uint32_t size, uint32_t id = 0)
+	virtual void addItem(const system::path& fullPath, uint32_t offset, uint32_t size, bool isDirectory, uint32_t id = 0)
 	{
 		SFileListEntry entry;
 		entry.ID = id ? id : m_files.size();
 		entry.offset = offset;
 		entry.size = size;
 		entry.name = fullPath;
-		entry.isDirectory = std::filesystem::is_directory(fullPath);
+		entry.isDirectory = isDirectory;
```

Comment:
why do you even need to keep track of directories in an archive?

Reply:
@sadiuk  ?